adagrams/game.py

```diff
@@ -1,11 +1,64 @@
+import random
+import copy
+
+LETTER_VALS = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1,
+        'S': 1, 'T': 1, 'D': 2, 'G': 2, 'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4, 'K': 5, 'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+
 def draw_letters():
-    pass
+    letter_pool = ['A'] * 9 + ['B'] * 2 + ['C'] * 2 + ['D'] * 4 + ['E'] * 12 + ['F'] * 2\
+    + ['G'] * 3 + ['H'] * 2 + ['I'] * 9 + ['J'] * 1 + ['K'] * 1 + ['L'] * 4 + ['M'] * 2\
+    + ['N'] * 6 + ['O'] * 8 + ['P'] * 2 + ['Q'] * 1 + ['R'] * 6 + ['S'] * 4 + ['T'] * 6\
+    + ['U'] * 4 + ['V'] * 2 + ['W'] * 2 + ['X'] * 1 + ['Y'] * 2 + ['Z'] * 1
+
+    return random.sample(letter_pool, 10)
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letters_available = copy.deepcopy(letter_bank)
+    for letter in word.upper():
+        if letter in letters_available:
+            letters_available.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    # Initialize the result to 0
+    result = 0
+
+    # Loop through the characters in the word
+    for character in word.upper(): 
+        # Check if the character is in the LETTER_VALS dictionary
+        if character in LETTER_VALS: 
+            # Add the value of the letter to the result
+            result += LETTER_VALS[character]
+
+    # If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+    if len(word) in range(7, 11):
+        result += 8
+
+    # Return the result
+    return result
+    
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_score = 0
+    winning_word = ''
+    for word in word_list:
+        word_score = score_word(word)
+        if highest_score < word_score:
+            highest_score = word_score
+            winning_word = word
+        elif highest_score == word_score and len(word) == 10 and len(winning_word) != 10: 
+            winning_word = word
+        elif highest_score == word_score and len(word) < len(winning_word) and len(winning_word) != 10:
+            winning_word = word
```

Comment:
This works great! Consider other ways this might be structured as well. Would having some of the if statements be nested improve or worsen the readability in your opinion? It's subjective though, there is no perfect answer!